WordPress/Classes/ViewRelated/Pages/ParentPagesController.swift

```diff
@@ -0,0 +1,69 @@
+import Foundation
+import CoreData
+
+/// Provides an API to access the current parent of a page, and list of parent pages available for editing.
+///
+@objc class ParentPagesController: NSObject {
```

Comment:
I needed a way to access the parent pages that can be selected for a child page. And to display them later in ["Set Page"](https://share.cleanshot.com/itDiZg) screen.

The good is that this logic is already implemented in `PageListTableViewHandler`. The bad news is that the class is [tightly coupled](https://share.cleanshot.com/yNpNpY) with `PageListViewController`'s table view. So I couldn't re-use that class elsewhere.

The solution was to read `PageListTableViewHandler` and extract only the code that fetches a page's parent pages available for editing. Then put in a new class that I called `ParentPagesController`. This class doesn't depend on any view and be used anywhere.

You can see how I use this class in `PageSettingsViewController`.

Let me know what you think. 🙏